src/engine/source/wdb/src/wdb.cpp

```diff
@@ -0,0 +1,195 @@
+#include <wdb/wdb.hpp>
+
+#include <iostream>
+#include <unistd.h>
+
+#include <logging/logging.hpp>
+
+#include "unixSocketInterface.hpp"
+
+namespace wazuhdb
+{
+
+WazuhDB::~WazuhDB()
+{
+    if (this->m_fd > 0)
+    {
+        close(this->m_fd);
+    }
+};
+
+void WazuhDB::connect()
+{
+    if (std::filesystem::exists(m_path) == false)
+    {
+        const std::string msg {"The wdb socket does not exist:" + m_path.string()};
+        throw std::runtime_error(msg);
+    }
+    else if (std::filesystem::is_socket(m_path) == false)
+    {
+        const std::string msg {"The wdb socket does not a socket:" + m_path.string()};
```

Comment:
"... is not a socket" or " does not refer to a socket"

Reply:
Yes